src/app/events/[event]/page.tsx

```diff
@@ -59,6 +67,17 @@ export default function Event({ params }: EventProps) {
     },
   })
 
+  const registerParticipantMutation = useMutation({
+    mutationFn: registerParticipant,
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ["participants", { event }] })
+    },
+  })
+
+  const handleRegisterParticipant = async (data: RegisterParticipant) => {
+    registerParticipantMutation.mutateAsync({ data, event })
```

Comment:
You're not handling the error here, so you can just use `mutate` instead of `mutateAsync`.